typescript/utils/src/altvm.ts

```diff
@@ -0,0 +1,657 @@
+import { MINIMUM_GAS } from './mingas.js';
+import { ProtocolType } from './types.js';
+
+// ### QUERY BASE ###
+export type ReqGetBalance = { address: string; denom: string };
+
+export type ReqGetTotalSupply = { denom: string };
+
+export type ReqEstimateTransactionFee<T> = {
+  transaction: T;
+  estimatedGasPrice: string;
+  sender: string;
+  senderPubKey?: string;
+  memo?: string;
+};
+export type ResEstimateTransactionFee = {
+  gasUnits: number;
+  gasPrice: number;
+  fee: number;
+};
+
+// ### QUERY CORE ###
+
+export type ReqGetMailbox = { mailboxId: string };
+export type ResGetMailbox = {
+  address: string;
+  owner: string;
+  localDomain: number;
+  defaultIsm: string;
+  defaultHook: string;
+  requiredHook: string;
+  messageSent: number;
+  messageReceived: number;
+};
```

Comment:
What is the difference between this and the derived config type? that we already have defined in the sdk?